module7/Homework7-start/Homework7/Models/UserData.swift

```diff
@@ -0,0 +1,214 @@
+/// Copyright (c) 2024 Kodeco Inc.
+/// 
+/// Permission is hereby granted, free of charge, to any person obtaining a copy
+/// of this software and associated documentation files (the "Software"), to deal
+/// in the Software without restriction, including without limitation the rights
+/// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+/// copies of the Software, and to permit persons to whom the Software is
+/// furnished to do so, subject to the following conditions:
+/// 
+/// The above copyright notice and this permission notice shall be included in
+/// all copies or substantial portions of the Software.
+/// 
+/// Notwithstanding the foregoing, you may not use, copy, modify, merge, publish,
+/// distribute, sublicense, create a derivative work, and/or sell copies of the
+/// Software in any work that is designed, intended, or marketed for pedagogical or
+/// instructional purposes related to programming, coding, application development,
+/// or information technology.  Permission for such use, copying, modification,
+/// merger, publication, distribution, sublicensing, creation of derivative works,
+/// or sale is expressly withheld.
+/// 
+/// This project and source code may use libraries or frameworks that are
+/// released under various Open-Source licenses. Use of those libraries and
+/// frameworks are governed by their own individual licenses.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+/// THE SOFTWARE.
+
+import Foundation
+
+struct UserData: Codable, Equatable {
+  let name: Name
+  let gender: String
+  let location: Location
+  let email: String
+  let login: Login
+  let dob: DOB
+  let registered: Registered
+  let phone: String
+  let cell: String
+  let id: ID
+  let picture: Picture
+  let nat: String
+  
+  struct Name: Codable, Equatable {
+    let title: String
+    let first: String
+    let last: String
+    
+    var fullName: String {
+      get {
+        title + " " + first + " " + last
+      }
+    }
+  }
+  
+  struct Location: Codable, Equatable {
+    let street: Street
+    let city: String
+    let state: String
+    let country: String
+    let postcode: Int // Note these can be strings as well - how to parse both?
+    let coordinates: LatLng
+    let timezone: Timezone
+        
+    struct Street: Codable, Equatable {
+      let number: Int
```

Comment:
What about "302A Main St." or "592 101st St"

Reply:
Same as above. Try parsing as both an Int and a String, store as a String, and then write back out as a String.